litellm/proxy/guardrails/guardrail_hooks/acuvity.py

```diff
@@ -0,0 +1,173 @@
+                self.post_guard_config = GuardConfig(self.post_guard_config_dict)
+        except Exception as e:
+            raise ValueError("Acuvity guard config cannot be parsed") from e
+
+        self.acuvity_client = Acuvity(Security(token=(api_key or os.environ["ACUVITY_TOKEN"])))
+        super().__init__(**kwargs)
+
+    @log_guardrail_information
+    async def async_pre_call_hook(
+        self,
+        user_api_key_dict: UserAPIKeyAuth,
+        cache: DualCache,
+        data: dict,
+        call_type: Literal[
+            "completion",
+            "text_completion",
+            "embeddings",
+            "image_generation",
+            "moderation",
+            "audio_transcription",
+            "pass_through_endpoint",
+            "rerank",
+        ],
+    ) -> Optional[Union[Exception, str, dict]]:
+        """
+        Runs before the LLM API call
+        Runs on only Input
+        We just update the data here if there is a pii detection else if more detections are deteected then we raise exception.
+        """
+
+        _messages = data.get("messages")
+        if _messages:
+            msgs = [message.get("content") for message in _messages if message.get("content") is not None]
+
+            verbose_proxy_logger.debug("Acuvity pre_call data: %s", msgs)
+            resp = await self.acuvity_client.apex.scan_async(*msgs, guard_config=self.pre_guard_config_dict)
+
+            # List of PII detectors to exclude from validation
+            pii_detectors = [
+                match
+                for detector in [
+                resp.guard_match(guard=GuardName.PII_DETECTOR),
+                resp.guard_match(guard=GuardName.SECRETS_DETECTOR),
+                resp.guard_match(guard=GuardName.KEYWORD_DETECTOR)
+                ]
+                for match in detector
+                if match.response_match == ResponseMatch.YES
+            ]
+
+            for index, extraction in enumerate(resp.scan_response.extractions):
+                # Skip if no match
+                if resp.match_details[index].response_match != ResponseMatch.YES:
+                    continue
+
+                # Check for violations
+                matched_checks = resp.match_details[index].matched_checks
+                if len(matched_checks) > len(pii_detectors):
+                    # Found violation beyond basic detectors
+                    violation = matched_checks[0].to_dict()
+                    raise HTTPException(
+                        status_code=400,
+                        detail={
+                            "error": "Violated guardrail policy",
+                            "guard": violation
+                        }
+                    )
+
+                if isinstance(_messages[index]["content"], str):
+                    _messages[index]["content"] = extraction.data
+
+                data["messages"] = _messages
```

Comment:
I think we should add a log message which states if we actually did redact any content or not. Because that's what this hook is supposed to be doing.